ui/src/dashboards/utils/protoboardToDashboard.ts

```diff
@@ -0,0 +1,112 @@
+import _ from 'lodash'
+
+import {getNextAvailablePosition} from 'src/dashboards/utils/cellGetters'
+
+import {
+  PBCell,
+  Protoboard,
+  Dashboard,
+  Template,
+  Source,
+  TemplateType,
+} from 'src/types'
+
+interface PBQueries {
+  query: string
+  groupbys: string[]
+  label: string
+}
+
+const addNewCellToCells = (
+  cells: Array<Partial<PBCell>>,
+  cell: Partial<PBCell>
+): Array<Partial<PBCell>> => {
+  let x = 0
+  let y = 0
+
+  if (cells.length !== 0) {
+    const pos = getNextAvailablePosition(cells, cell)
+    x = pos.x
+    y = pos.y
+  }
+
+  return [
+    ...cells,
+    {
+      ...cell,
+      x,
+      y,
+    },
+  ]
+}
+
+const collision = (
```

Comment:
how do you feel about making this func name `isCollision` or something that indicates it returns a boolean?